registry/auth/token/token.go

```diff
@@ -172,26 +172,27 @@ func (t *Token) VerifySigningKey(verifyOpts VerifyOptions) (signingKey crypto.Pu
 	// verifying the first one in the list only at the moment.
 	header := t.JWT.Headers[0]
 
-	signingKey, err = verifyCertChain(header, verifyOpts.Roots)
-	// NOTE(milosgajdos): if the x5c header is missing
-	// the token may have been signed by a JWKS.
-	if err != nil && err != jose.ErrMissingX5cHeader {
-		return
-	}
-
-	switch {
-	case header.JSONWebKey != nil:
-		signingKey, err = verifyJWK(header, verifyOpts)
-	case len(header.KeyID) > 0:
-		signingKey = verifyOpts.TrustedKeys[header.KeyID]
-		if signingKey == nil {
-			err = fmt.Errorf("token signed by untrusted key with ID: %q", header.KeyID)
+	signingKey, err := verifyCertChain(header, verifyOpts.Roots)
+	if err != nil {
+		// NOTE(milosgajdos): if the x5c header is missing
+		// the token may have been signed by a JWKS.
+		if errors.Is(err, jose.ErrMissingX5cHeader) {
+			switch {
+			case header.JSONWebKey != nil:
+				return verifyJWK(header, verifyOpts)
+			case len(header.KeyID) > 0:
```

Comment:
nit: why are we calling `len` on a string when all were doing is checking for it to be populated?

```suggestion
			case header.KeyID != "":
```

Reply:
I think it's mostly an old habit; if I'm checking if the string is empty I use the `>` operator; if I'm checking for a specific string I use the `==| !=` operator. I think in the past Go compiler generated different code but I don't think it's the case anymore; you can see both cases in std library.